govuk-one-login/authentication-frontend
# AUT-1712: Remove Zendesk and previous Contact Centre constraints

## What?

Removes code related to Zendesk support and constraints associated with previous Contact Centre on One Login contact forms.

## Why?

The new Contact Centre and SmartAgent are live in production, and we no longer need to support dual-running of Zendesk and the previous Contact Centre.

## Manual testing of forms

While the SmartAgent integration should not be impacted by these changes, the code changes are so broad ranging that I've also manually tested the following sample of forms from my local machine to verify: content and error messages are rendered in the correct language; users are presented with the "Your message has been submitted" message upon submission and; submissions are being routed to the SmartAgent UAT environment (which is where my local development machine is configured to route submissions).

| Form path | Displays English content - `with(out)?` validation errors | Displays Welsh content - `with(out)?` validation errors | "Your message has been submitted" rendered | SmartAgent submission received (English) | SmartAgent submission received (Welsh)
|    :----:   |    :----:   |    :----:   |    :----:   |    :----:   |    :----:   |
| `/contact-us-questions?theme=account_creation&subtheme=no_security_code` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=signing_in&subtheme=account_not_found` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=something_else` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=id_check_app&subtheme=linking_problem` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=id_face_to_face&subtheme=face_to_face_details` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=proving_identity` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=email_subscriptions` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=suggestions_feedback` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=account_creation&subtheme=technical_error` | ✅ | ✅ | ✅ | ✅ | ✅ |
| `/contact-us-questions?theme=id_face_to_face&subtheme=face_to_face_post_office_service` | ✅ | ✅ | ✅ | ✅ | ✅ |
 
I have also checked with QA to verify that this change will not impact our acceptance tests.
